miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/WebConfig.java

```diff
@@ -55,6 +66,16 @@ public RequestMappingHandlerAdapter requestMappingHandlerAdapter() {
     return adapter;
   }
 
+  @Value("${miso.run.report.links:}")
+  private String runReportLinksConfigLine;
+
+  @Bean
+  public ExternalUriBuilder externalUriBuilder() {
+    ExternalUriBuilder externalUriBuilder = new ExternalUriBuilder();
+    externalUriBuilder.setRunReportLinksConfig(runReportLinksConfigLine);
+    return externalUriBuilder;
+  }
+
   @Autowired
   private List<BarcodableService> barcodableServices;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Inject this field value directly into "barcodableServicesMap", the only method that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3305)